planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp

```diff
@@ -2379,23 +2398,19 @@ void AvoidanceModule::addShiftLineIfApproved(const AvoidLineArray & shift_lines)
     const size_t prev_size = path_shifter_.getShiftLinesSize();
     addNewShiftLines(path_shifter_, shift_lines);
 
+    current_raw_shift_lines_ = avoidance_data_.unapproved_raw_sl;
+
     // register original points for consistency
     registerRawShiftLines(shift_lines);
 
-    int i = shift_lines.size() - 1;
-    for (; i > 0; i--) {
-      if (fabs(shift_lines.at(i).getRelativeLength()) < 0.01) {
-        continue;
-      } else {
-        break;
-      }
-    }
+    const auto sl = getNotStraightShiftLine(shift_lines);
+    const auto sl_front = shift_lines.front();
+    const auto sl_back = shift_lines.back();
 
-    if (shift_lines.at(i).getRelativeLength() > 0.0) {
-      left_shift_array_.push_back({uuid_left_, shift_lines.front().start, shift_lines.back().end});
-    } else if (shift_lines.at(i).getRelativeLength() < 0.0) {
-      right_shift_array_.push_back(
-        {uuid_right_, shift_lines.front().start, shift_lines.back().end});
+    if (sl.getRelativeLength() > 0.0) {
+      left_shift_array_.push_back({uuid_left_, sl_front.start, sl_back.end});
```

Comment:
`emplace_back` is preferred here and next line. (This is not a part of PR, you don't need to change though)

Reply:
I'll fix in another PR. Thanks.